common/test/acceptance/tests/helpers.py

```diff
@@ -354,6 +359,92 @@ def verify_events_of_type(self, username, event_type, expected_events, expected_
                     "Refer '{0}' does not have correct suffix, '{1}'.".format(actual_referer, expected_referers[index])
                 )
 
+    @contextmanager
+    def capture_events(self, event_filter=None, number_of_matches=1, captured_events=None):
+        start_time = datetime.utcnow()
+
+        yield
+
+        events = self.wait_for_events(
+            start_time=start_time, event_filter=event_filter, number_of_matches=number_of_matches)
+
+        if captured_events is not None and hasattr(captured_events, 'append') and callable(captured_events.append):
+            for event in events:
+                captured_events.append(event)
+
+    def wait_for_events(self, start_time=None, event_filter=None, number_of_matches=1):
```

Comment:
I think, going with bok-choy patterns, that this wait_for should have a timeout setting that you can pass down to the Promise.